src/app/shared/zap-dialog/zap-dialog.component.ts

```diff
@@ -0,0 +1,223 @@
+import { ZapQrCodeComponent } from '../zap-qr-code/zap-qr-code.component';
+import { NostrProfileDocument, LNURLPayRequest, LNURLInvoice, NostrEventDocument } from 'src/app/services/interfaces';
+
+export interface ZapDialogData {
+  profile: NostrProfileDocument;
+}
+
+@Component({
+  selector: 'app-zap-dialog',
+  templateUrl: './zap-dialog.component.html',
+  styleUrls: ['./zap-dialog.component.scss']
+})
+export class ZapDialogComponent implements OnInit  {
+  sendZapForm!: UntypedFormGroup;
+  minSendable: number = 0;
+  maxSendable: number = 0;
+  profile!: NostrProfileDocument
+  amount: number = 0;
+  comment = "";
+  payRequest: LNURLPayRequest | null = null
+  invoice: LNURLInvoice = {
+    pr: ""
+  }
+
+  imagePath = '/assets/profile.png';
+  tooltip = '';
+  tooltipName = '';
+  profileName = '';
+  error: string = "";
+
+  constructor(@Inject(MAT_DIALOG_DATA) public data: ZapDialogData, 
+    private formBuilder: UntypedFormBuilder,
+    private eventService: EventService,
+    private relayService: RelayService,
+    private nostr: NostrService,
+    private util: Utilities,
+    private dataService: DataService,
+    private dialog: MatDialog,
+    public dialogRef: MatDialogRef<ZapDialogComponent>
+  ) {
+
+
+  }
+
+  async ngOnInit() {
+    this.profile = this.data.profile
+    this.sendZapForm = this.formBuilder.group({
+      amount: ['', [Validators.required]],
+      comment: ['']
+    })
+
+    this.fetchPayReq()
+    await this.updateProfileDetails();
+  }
+
+  async fetchPayReq(): Promise<void> {
+    this.payRequest = await this.fetchZapper() 
+    this.recofigureFormValidators()
+  }
+
+
+  async fetchZapper(): Promise<LNURLPayRequest | null> {
+    let staticPayReq = ""
+    if(this.profile.lud16) {
+      const parts = this.profile.lud16.split("@")
+      staticPayReq = `https://${parts[1]}/.well-known/lnurlp/${parts[0]}`;
+    } else if (this.profile.lud06 && this.profile.lud06.toLowerCase().startsWith("lnurl")) {
+      staticPayReq = this.util.convertBech32ToText(this.profile.lud06).toString()
+    }
+
+    if(staticPayReq.length !== 0) {
+      try {
+        const resp = await fetch(staticPayReq);
+        if(resp.ok) {
+          const payReq = await resp.json();
+          if(payReq.status === "ERROR") {
+            this.error = payReq.reason ? payReq.reason : "Error fetching the invoice - please try again later"
+          } else {
+            return payReq
+          }
+        }
+      } catch (err) {
+        this.error = "Error fetching the invoice - please try again later"
+      }
+    }
+
+    return null
+  }
+
+  async onSubmit() {
+    if (this.sendZapForm.valid) {
+      let comment = this.sendZapForm.get('comment')?.value
+      let amount = this.sendZapForm.get('amount')?.value
+      if(!amount || !this.payRequest) {
+        console.log( "error: please enter an amount and  a valid pay request")
+      } else {
+        const callback = new URL(this.payRequest.callback)
```

Comment:
I don't know what editor you use, but a tip for the future: ALT-SHIFT-F will auto-format your entire source file, appending things like semicolon at the ends and it will do this in accordance with the editor rules set for the whole project.